### PR TITLE
Gaussian shuffle and tutorials

### PR DESCRIPTION
**Pull request by gmischl1**
#### What does this implement/fix? Explain your changes.
This PR adds shuffle and random_state setting ability to the GaussianMixtures class for synthetic data generation. It also updates the gaussian mixture tutorial, and uses the gaussian mixtures in a dcca tutorial notebook.

#### Any other comments?
It also fixes minor documentation errors like missing Attributes section in GaussianMixtures

### CODE REVIEW

File: mvlearn/datasets/gaussian_mixture.py [hunk 1]
```diff
@@ -139,6 +181,12 @@ def __init__(self, mu, sigma, n, class_probs=None):
                 ]
             )
 
+        if self.shuffle:
+            np.random.seed(self.shuffle_random_state)
```
Comment (rflperry) [inline mvlearn/datasets/gaussian_mixture.py:185]: Correct me but I believe this would shuffle y and latent independently if shuffle_random_state is None.

Comment (rflperry) [reply mvlearn/datasets/gaussian_mixture.py:185]: Shuffle a list of indices and apply to both

**Comment by rflperry** [comment]
There appears to be a conflict too, with docs/tutorials.rst

File: mvlearn/datasets/gaussian_mixture.py (hunk 1 above)
Comment (gmischl1) [reply mvlearn/datasets/gaussian_mixture.py:185]: I also set the random seed before I shuffle the labels below though. I plotted both and I believe they are shuffled correctly.

**Comment by gmischl1** [comment]
Fixed issue with seeding for shuffle.

**Pull request merged**